src/controllers/mainController.ts

```diff
@@ -1411,4 +1422,41 @@ export default class MainController implements vscode.Disposable {
 	public onClearAzureTokenCache(): void {
 		this.connectionManager.onClearTokenCache();
 	}
+
+	private ExecutionPlanCustomEditorProvider = class implements vscode.CustomTextEditorProvider {
+		context: vscode.ExtensionContext;
+		executionPlanService: ExecutionPlanService;
+		untitledSqlService: UntitledSqlDocumentService
+        constructor(
+			context: vscode.ExtensionContext,
+			executionPlanService: ExecutionPlanService,
+			untitledSqlService: UntitledSqlDocumentService
+        ) {
+			this.context = context;
+			this.executionPlanService = executionPlanService;
+			this.untitledSqlService = untitledSqlService;
+        }
+
+        public async resolveCustomTextEditor(
+            document: vscode.TextDocument
+        ): Promise<void> {
+			await this.onOpenExecutionPlanFile(document);
+        }
+
+		public async onOpenExecutionPlanFile(document: vscode.TextDocument) {
+			const planContents = document.getText();
+			let docName = document.fileName;
+			docName = docName.substring(docName.lastIndexOf(path.sep) + 1);
+
+			const executionPlanPanel = new ExecutionPlanWebViewController(
```

Comment:
nit: why is a controller object named `executionPlanPanel`, shouldn't it be something like `executionPlanController`?